npm/src/controller/sso-handler.ts

```diff
@@ -208,9 +208,60 @@ export class SAMLHandler {
     };
   }
 
-  createSAMLResponse = async (params: { profile: SAMLProfile; session: any }) => {
-    const { profile, session } = params;
+  async createOIDCRequest({
+    connection,
+    requestParams,
+  }: {
+    connection: OIDCSSORecord;
+    requestParams: Record<string, any>;
+  }) {
+    if (!this.opts.oidcPath) {
+      throw new JacksonError('OpenID response handler path (oidcPath) is not set', 400);
+    }
+
+    const { discoveryUrl, metadata, clientId, clientSecret } = connection.oidcProvider;
+
+    try {
+      const oidcIssuer = await oidcIssuerInstance(discoveryUrl, metadata);
+      const oidcClient = new oidcIssuer.Client({
+        client_id: clientId!,
+        client_secret: clientSecret,
+        redirect_uris: [this.opts.externalUrl + this.opts.oidcPath],
+        response_types: ['code'],
+      });
+
+      const oidcCodeVerifier = generators.codeVerifier();
+      const code_challenge = generators.codeChallenge(oidcCodeVerifier);
+      const oidcNonce = generators.nonce();
+      const scope = ['openid', 'email', 'profile']
+        .filter((value, index, self) => self.indexOf(value) === index)
```

Comment:
We don't need filter here as it's a static list and not dependent on request params